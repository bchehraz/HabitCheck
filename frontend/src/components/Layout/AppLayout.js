import React from "react"
import PropTypes from "prop-types"
import { ThemeProvider } from "styled-components"

import { AppHeader, FooterNav } from "./"
import "../Landing/layout.css"
import "./layout.scss"

const AppLayout = ({ children, path }) => {
  let page = path.replace("/app/", "")
  if (page === "/app" || page === "") {
    page = "Today"
  } else if (page === "settings") {
    page = "Settings"
  } else {
    page = "Journal"
  }
  return (
    <>
      <ThemeProvider theme={{ mode: "light" }}>
        <AppHeader title={page} />
        <div
          style={{
            margin: `0 auto 5rem`,
            padding: 0,
            maxWidth: "100%",
            display: "flex",
            flexFlow: "column nowrap",
            alignItems: "center",
            position: "relative",
          }}
        >
          <main>{children}</main>
        </div>
        <FooterNav currentPage={page} path={path} />
      </ThemeProvider>
    </>
  )
}

AppLayout.propTypes = {
  children: PropTypes.node.isRequired,
  path: PropTypes.string.isRequired,
}

export default AppLayout
