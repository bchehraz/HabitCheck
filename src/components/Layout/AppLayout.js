import React from "react"
import PropTypes from "prop-types"

import { AppHeader, FooterNav } from "./"
import "../landing/layout.css"
import "./layout.scss"

const AppLayout = ({ children, path }) => {
  let page = path.replace("/app/", "")
  if (page === "/app" || page === "") {
    page = "Today"
  } else if (page === "settings") {
    page = "Settings"
  } else {
    page = "Your Journal"
  }
  return (
    <>
      <AppHeader title={page} />
      <div
        style={{
          margin: `0 auto 5rem`,
          padding: 0,
          maxWidth: "100%",
          display: "flex",
          flexFlow: "column nowrap",
          alignItems: "center",
        }}
      >
        <main>{children}</main>
      </div>
      <FooterNav currentPage={page} />
    </>
  )
}

AppLayout.propTypes = {
  children: PropTypes.node.isRequired,
  path: PropTypes.string.isRequired,
}

export default AppLayout
