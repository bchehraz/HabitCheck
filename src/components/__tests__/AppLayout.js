import React from "react"
import renderer from "react-test-renderer"
import { advanceTo, clear } from "jest-date-mock"

import AppLayout from "../AppLayout"

describe("Header", () => {
  it("renders correctly with /app/ path", () => {
    advanceTo(new Date("12/25/2019"))
    const tree = renderer
      .create(
        <AppLayout path="/app/">
          <h1>Some Content</h1>
        </AppLayout>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
    clear()
  })

  it("renders correctly with /app/stats", () => {
    advanceTo(new Date("12/25/2019"))
    const tree = renderer
      .create(
        <AppLayout path="/app/stats">
          <h1>Some Content</h1>
        </AppLayout>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
    clear()
  })

  it("renders correctly with /app/journal", () => {
    advanceTo(new Date("12/25/2019"))
    const tree = renderer
      .create(
        <AppLayout path="/app/journal">
          <h1>Some Content</h1>
        </AppLayout>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
    clear()
  })
})
