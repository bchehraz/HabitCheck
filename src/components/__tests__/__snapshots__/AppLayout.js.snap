// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly with /app/ path 1`] = `
Array [
  <header>
    <div
      className="sc-bdVaJa bzVTIk"
    >
      <h1>
        Today
      </h1>
      <p>
        Wednesday, December 25 2019
      </p>
    </div>
  </header>,
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexFlow": "column nowrap",
        "margin": "0 auto 5rem",
        "maxWidth": "100%",
        "padding": 0,
      }
    }
  >
    <main>
      <h1>
        Some Content
      </h1>
    </main>
  </div>,
  <div
    className="sc-bwzfXH hCiwAb"
  >
    <div
      className="sc-htpNat hMjIwR footer"
    >
      <a
        href="/app/stats"
      >
        <div
          className="sc-bxivhb dQxvwo habitStats"
        >
          <svg
            fill="currentColor"
            height={32}
            size={32}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width={32}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M500 384c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H12c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v308h436zM372.7 159.5L288 216l-85.3-113.7c-5.1-6.8-15.5-6.3-19.9 1L96 248v104h384l-89.9-187.8c-3.2-6.5-11.4-8.7-17.4-4.7z"
            />
          </svg>
        </div>
      </a>
      <a
        href="/app/"
      >
        <div
          className="sc-bxivhb bEdBIT today"
        >
          <div
            className="sc-ifAKCX biWapd"
          />
          <svg
            className="todayIcon"
            fill="none"
            height={40}
            size={40}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width={40}
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="9 11 12 14 22 4"
            />
            <path
              d="M21 12v7a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h11"
            />
          </svg>
        </div>
      </a>
      <a
        href="/app/journal"
      >
        <div
          className="sc-bxivhb dQxvwo yourJournal"
        >
          <svg
            className="journalIcon"
            fill="currentColor"
            height={32}
            size={32}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 576 512"
            width={32}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M542.22 32.05c-54.8 3.11-163.72 14.43-230.96 55.59-4.64 2.84-7.27 7.89-7.27 13.17v363.87c0 11.55 12.63 18.85 23.28 13.49 69.18-34.82 169.23-44.32 218.7-46.92 16.89-.89 30.02-14.43 30.02-30.66V62.75c.01-17.71-15.35-31.74-33.77-30.7zM264.73 87.64C197.5 46.48 88.58 35.17 33.78 32.05 15.36 31.01 0 45.04 0 62.75V400.6c0 16.24 13.13 29.78 30.02 30.66 49.49 2.6 149.59 12.11 218.77 46.95 10.62 5.35 23.21-1.94 23.21-13.46V100.63c0-5.29-2.62-10.14-7.27-12.99z"
            />
          </svg>
        </div>
      </a>
    </div>
  </div>,
]
`;

exports[`Header renders correctly with /app/journal 1`] = `
Array [
  <header>
    <div
      className="sc-bdVaJa bzVTIk"
    >
      <h1>
        Your Journal
      </h1>
      <p>
        Wednesday, December 25 2019
      </p>
    </div>
  </header>,
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexFlow": "column nowrap",
        "margin": "0 auto 5rem",
        "maxWidth": "100%",
        "padding": 0,
      }
    }
  >
    <main>
      <h1>
        Some Content
      </h1>
    </main>
  </div>,
  <div
    className="sc-bwzfXH hCiwAb"
  >
    <div
      className="sc-htpNat hMjIwR footer"
    >
      <a
        href="/app/stats"
      >
        <div
          className="sc-bxivhb dQxvwo habitStats"
        >
          <svg
            fill="currentColor"
            height={32}
            size={32}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width={32}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M500 384c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H12c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v308h436zM372.7 159.5L288 216l-85.3-113.7c-5.1-6.8-15.5-6.3-19.9 1L96 248v104h384l-89.9-187.8c-3.2-6.5-11.4-8.7-17.4-4.7z"
            />
          </svg>
        </div>
      </a>
      <a
        href="/app/"
      >
        <div
          className="sc-bxivhb dQxvwo today"
        >
          <svg
            className="todayIcon"
            fill="none"
            height={32}
            size={32}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width={32}
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="9 11 12 14 22 4"
            />
            <path
              d="M21 12v7a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h11"
            />
          </svg>
        </div>
      </a>
      <a
        href="/app/journal"
      >
        <div
          className="sc-bxivhb bEdBIT yourJournal"
        >
          <div
            className="sc-ifAKCX biWapd"
          />
          <svg
            className="journalIcon"
            fill="currentColor"
            height={40}
            size={40}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 576 512"
            width={40}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M542.22 32.05c-54.8 3.11-163.72 14.43-230.96 55.59-4.64 2.84-7.27 7.89-7.27 13.17v363.87c0 11.55 12.63 18.85 23.28 13.49 69.18-34.82 169.23-44.32 218.7-46.92 16.89-.89 30.02-14.43 30.02-30.66V62.75c.01-17.71-15.35-31.74-33.77-30.7zM264.73 87.64C197.5 46.48 88.58 35.17 33.78 32.05 15.36 31.01 0 45.04 0 62.75V400.6c0 16.24 13.13 29.78 30.02 30.66 49.49 2.6 149.59 12.11 218.77 46.95 10.62 5.35 23.21-1.94 23.21-13.46V100.63c0-5.29-2.62-10.14-7.27-12.99z"
            />
          </svg>
        </div>
      </a>
    </div>
  </div>,
]
`;

exports[`Header renders correctly with /app/stats 1`] = `
Array [
  <header>
    <div
      className="sc-bdVaJa bzVTIk"
    >
      <h1>
        Habit Stats
      </h1>
      <p>
        Wednesday, December 25 2019
      </p>
    </div>
  </header>,
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexFlow": "column nowrap",
        "margin": "0 auto 5rem",
        "maxWidth": "100%",
        "padding": 0,
      }
    }
  >
    <main>
      <h1>
        Some Content
      </h1>
    </main>
  </div>,
  <div
    className="sc-bwzfXH hCiwAb"
  >
    <div
      className="sc-htpNat hMjIwR footer"
    >
      <a
        href="/app/stats"
      >
        <div
          className="sc-bxivhb bEdBIT habitStats"
        >
          <div
            className="sc-ifAKCX biWapd"
          />
          <svg
            fill="currentColor"
            height={40}
            size={40}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width={40}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M500 384c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H12c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v308h436zM372.7 159.5L288 216l-85.3-113.7c-5.1-6.8-15.5-6.3-19.9 1L96 248v104h384l-89.9-187.8c-3.2-6.5-11.4-8.7-17.4-4.7z"
            />
          </svg>
        </div>
      </a>
      <a
        href="/app/"
      >
        <div
          className="sc-bxivhb dQxvwo today"
        >
          <svg
            className="todayIcon"
            fill="none"
            height={32}
            size={32}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width={32}
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="9 11 12 14 22 4"
            />
            <path
              d="M21 12v7a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h11"
            />
          </svg>
        </div>
      </a>
      <a
        href="/app/journal"
      >
        <div
          className="sc-bxivhb dQxvwo yourJournal"
        >
          <svg
            className="journalIcon"
            fill="currentColor"
            height={32}
            size={32}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 576 512"
            width={32}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M542.22 32.05c-54.8 3.11-163.72 14.43-230.96 55.59-4.64 2.84-7.27 7.89-7.27 13.17v363.87c0 11.55 12.63 18.85 23.28 13.49 69.18-34.82 169.23-44.32 218.7-46.92 16.89-.89 30.02-14.43 30.02-30.66V62.75c.01-17.71-15.35-31.74-33.77-30.7zM264.73 87.64C197.5 46.48 88.58 35.17 33.78 32.05 15.36 31.01 0 45.04 0 62.75V400.6c0 16.24 13.13 29.78 30.02 30.66 49.49 2.6 149.59 12.11 218.77 46.95 10.62 5.35 23.21-1.94 23.21-13.46V100.63c0-5.29-2.62-10.14-7.27-12.99z"
            />
          </svg>
        </div>
      </a>
    </div>
  </div>,
]
`;
